website/src/comment-panel.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {SetStateAction, ReactNode } from "react"
+import * as Dailp from "src/graphql/dailp"
+import { TranslatedParagraph } from "./segment"
+import { useState } from "react"
+import { Button } from "./components"
+import * as css from "./comment-panel.css"
+
+export const CommentPanel = (p: {
+    word: Dailp.FormFieldsFragment | null
+    segment: TranslatedParagraph | null
+    setCommentsPanel: React.Dispatch<SetStateAction<boolean>>
```

Comment:
The way I am doing this works, but also feels very iffy.

Reply:
left a comment about this elsewhere. To start, try
```
export const CommentPanel = (p: {
    parentText: Dailp.FormFieldsFragment | TranslatedParagraph  | null,
    setCommentsPanel: React.Dispatch<SetStateAction<boolean>>
```